packages/mobile/src/web3/dataEncryptionKey.ts

```diff
@@ -103,10 +126,53 @@ export function* registerAccountDek(account: string) {
     }
 
     // Generate and send a transaction to set the DEK on-chain.
-    const setAccountTx = accountsWrapper.setAccount('', publicDataKey, account)
+    let setAccountTx = accountsWrapper.setAccount('', publicDataKey, walletAddress)
     const context = newTransactionContext(TAG, 'Set wallet address & DEK')
-    yield call(sendTransaction, setAccountTx.txo, account, context)
+    // If MTW has been created, route the user's DEK/wallet registration through it
+    // because accountAddress is determined by msg.sender
+    if (mtwAddress) {
```

Comment:
Nit consider breaking this out. Similar to my comment above, it probably makes sense to group MTW related complexity in one place. And/or maybe a new option to `sendTransaction` that handles this re-routing. Ultimately up to you, just thinking out loud!

Reply:
Yeah, we are kinda caught in a weird spot where we want to eventually move towards every user having a MTW but first introduce it only for a specific use case (feeless onboarding). That's why I've been holding off on abstracting this kind of logic into functions that are widely used like `sendTransaction`.

Your point still stands though - I will think about how I can better contain this logic.